### PR TITLE
[CodeStyle] Update clang-format config to make it compatible with Lightning simulators

### PR DESCRIPTION
**Pull request by maliasadi**
Update clang-format config to make it compatible with Lightning simulators

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -1,8 +1 @@
 BasedOnStyle: LLVM
-UseTab: Never
-IndentWidth: 4
-TabWidth: 4
-AccessModifierOffset: -2
-BreakBeforeBraces: Stroustrup
-FixNamespaceComments: true
-ColumnLimit: 100
```
Comment (maliasadi) [inline .clang-format:8]: @mlxd we discuss this yesterday and here is a fix for the code style inconsistency. What do you think?

Comment (mlxd) [reply .clang-format:8]: `BreakBeforeBraces: Stroustrup`  There's the culprit. Sure, sounds good to me

Comment (maliasadi) [reply .clang-format:8]: Yep, there it is! :)

**Review by erick-xanadu — COMMENTED** [review]
I don't have strong opinions about style. Consider this comment as my approval, I just think that you should leave this PR open for longer so that other people to comment on it. Thanks!

**Comment by mlxd** [comment]
I'm curious about your thoughts on the other options: should the lightning repos match the 100 line limit. Since we do use this with PL and black, I'm not against sticking this in.

Also, how about the others?

**Comment by maliasadi** [comment]
@mlxd I was looking into Lightning simulators and I'm now wondering about the use case of our original style and if we can settle and agree on a unified clang-format config file. I want to root for the following options,
- `AccessModifierOffset: -2` : The extra indent or outdent of access modifiers. This seems to be more natural and that's actually one of the options followed in other C++ software.
- `FixNamespaceComments: true` : This adds missing namespace end comments for namespaces and fixes invalid existing ones. Super useful imho.
- `ColumnLimit: 100` : We had some discussions in the past that it should be 100 or 80. I "still" think 100 is more standard and should be part of the config file.

There may be other options worth exploring and added to the style. It's good to add a unified `.clang-format` to our C++ projects.

**Comment by mlxd** [comment]
> @mlxd I was looking into Lightning simulators and I'm now wondering about the use case of our original style and if we can settle and agree on a unified clang-format config file. I want to root for the following options,
> 
>     * `AccessModifierOffset: -2` : The extra indent or outdent of access modifiers. This seems to be more natural and that's actually one of the options followed in other C++ software.
> 
>     * `FixNamespaceComments: true` : This adds missing namespace end comments for namespaces and fixes invalid existing ones. Super useful imho.
> 
>     * `ColumnLimit: 100` : We had some discussions in the past that it should be 100 or 80. I "still" think 100 is more standard and should be part of the config file.
> 
> 
> There may be other options worth exploring and added to the style. It's good to add a unified `.clang-format` to our C++ projects.

Sounds good to me. Let's keep this open for now, and potentially offload all of these into a central repo for use across the ecosystem in Q3.

**Comment by maliasadi** [comment]
> I'm curious about your thoughts on the other options: should the lightning repos match the 100 line limit. Since we do use this with PL and black, I'm not against sticking this in. Also, how about the others?

Ahh.. I answered a few of these in my recent comment. Being said, I think we can keep those options and also add this `.clang-format` to Lightning simualtors (and any other Xanadu's C++ projects). What do you think?

**Comment by maliasadi** [comment]
> Let's keep this open for now, and potentially offload all of these into a central repo for use across the ecosystem in Q3.

Let's do this! Thanks! :)

**Comment by maliasadi** [comment]
> Let's keep this open for now, and potentially offload all of these into a central repo for use across the ecosystem in Q3.

Let's do this! :100:

**Comment by maliasadi** [comment]
Fixed all conflicts and updated the work in https://github.com/PennyLaneAI/catalyst/pull/295

**Pull request closed without merging**